tests/testCompoundCatalogDBObject.py

```diff
@@ -0,0 +1,667 @@
+        db1 = dbClass4(database=self.dbName, driver='sqlite')
+        db2 = dbClass5(database=self.dbName, driver='sqlite')
+        with self.assertRaises(RuntimeError) as context:
+            compound = CompoundCatalogDBObject([db1, db2])
+
+        self.assertTrue("objid class4 is duplicated" in context.exception.message)
+
+        # test case where CompoundCatalogDBObject does not support the
+        # tables being queried
+        db1 = dbClass1(database=self.dbName, driver='sqlite')
+        db2 = dbClass2(database=self.dbName, driver='sqlite')
+        with self.assertRaises(RuntimeError) as context:
+            compound = specificCompoundObj_otherTest([db1, db2])
+
+        msg = "This CompoundCatalogDBObject does not support the table 'test'"
+        self.assertTrue(msg in context.exception.message)
+
+
+
+    def testCompoundCatalogDBObject(self):
+        """
+        Verify that CompoundCatalogDBObject returns the expected
+        columns.
+        """
+        db1 = dbClass1(database=self.dbName, driver='sqlite')
+        db2 = dbClass2(database=self.dbName, driver='sqlite')
+        db3 = dbClass3(database=self.dbName, driver='sqlite')
+        dbList = [db1, db2, db3]
+        compoundDb = CompoundCatalogDBObject(dbList)
+
+        colNames = ['class1_aa', 'class1_bb',
+                    'class2_aa', 'class2_bb',
+                    'class3_aa', 'class3_bb', 'class3_cc']
+
+        results = compoundDb.query_columns(colnames=colNames)
+
+        for chunk in results:
+            numpy.testing.assert_array_almost_equal(chunk['class1_aa'],
+                                                    self.controlArray['a'],
+                                                    decimal=6)
+
+            numpy.testing.assert_array_equal(chunk['class1_bb'],
+                                             self.controlArray['d'])
+
+            numpy.testing.assert_array_almost_equal(chunk['class2_aa'],
+                                                    2.0*self.controlArray['b'],
+                                                    decimal=6)
+
+            numpy.testing.assert_array_almost_equal(chunk['class2_bb'],
+                                                    self.controlArray['a'],
+                                                    decimal=6)
+
+
+            numpy.testing.assert_array_almost_equal(chunk['class3_aa'],
+                                                    self.controlArray['c']-3.0,
+                                                    decimal=6)
+
+
+            numpy.testing.assert_array_almost_equal(chunk['class3_bb'],
+                                                    self.controlArray['a'],
+                                                    decimal=6)
+
+            numpy.testing.assert_array_almost_equal(chunk['class3_cc'],
+                                                    3.0*self.controlArray['b'],
+                                                    decimal=6)
+
```

Comment:
Should there also be some sort of test for the default values from each catalogDBObject?